### PR TITLE
Label 4a

### PR DESCRIPTION
**Pull request by rpatel3001**

### CODE REVIEW

**Comment by kevinelliott** [comment]
Holy batman, @rpatel3001 ... huge PR, nice...

**Comment by rpatel3001** [comment]
haha probably should have split the non 4A stuff out, my bad. still need to add a couple tests for 4A that I forgot btw

File: lib/plugins/Label_21_POS.ts [hunk 1]
```diff
@@ -31,7 +31,7 @@ export class Label_21_POS extends DecoderPlugin {
       processPosition(decodeResult, fields[0].trim());
       ResultFormatter.time_of_day(decodeResult, DateTimeUtils.convertHHMMSSToTod(fields[2]));
       ResultFormatter.altitude( decodeResult, Number(fields[3]));
-      ResultFormatter.temperature(decodeResult, fields[6]);
+      ResultFormatter.temperature(decodeResult, fields[6].replace(/ /g, ""));
```
Comment (makrsmark) [inline lib/plugins/Label_21_POS.ts:34]: Could replace with `0` but this works the same

File: lib/plugins/Label_4A.test.ts [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+  expect(decoderPlugin.name).toBe('label-4a');
+  expect(decoderPlugin.qualifiers).toBeDefined();
+  expect(decoderPlugin.qualifiers()).toEqual({
+    labels: ['4A'],
+  });
+});
+
+test('decodes Label 4A, variant 1', () => {
+  const decoder = new MessageDecoder();
+  const decoderPlugin = new Label_4A(decoder);
```
Comment (makrsmark) [inline lib/plugins/Label_4A.test.ts:18]: This is me being opinionated. Can we wrap all the tests in a `Label 4A` describe and have this be in a beforeEach (all?) so it doesn't need to be in every test?

File: lib/plugins/Label_4A.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  }
+
+  decode(message: Message, options: Options = {}) : DecodeResult {
+    const decodeResult = this.defaultResult();
+    decodeResult.decoder.name = this.name;
+    decodeResult.message = message;
+    decodeResult.formatted.description = 'Latest New Format';
+
+
+    // Inmarsat C-band seems to prefix normal messages with a message number and flight number
```
Comment (makrsmark) [inline lib/plugins/Label_4A.ts:24]: I think I've seen this before. Is it worth a function?

File: lib/plugins/Label_4A.test.ts [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+  expect(decodeResult.formatted.items[1].code).toBe('ETA');
+  expect(decodeResult.formatted.items[1].value).toBe('13:20:00');
+  expect(decodeResult.formatted.items[2].code).toBe('ALT');
+  expect(decodeResult.formatted.items[2].value).toBe('33467 feet');
+  expect(decodeResult.formatted.items[3].code).toBe('POS');
+  expect(decodeResult.formatted.items[3].value).toBe('41.093 N, 72.677 W');
+});
+
+// disabled because all messages should decode
+test('decodes Label 4A <invalid>', () => {
```
Comment (makrsmark) [inline lib/plugins/Label_4A.test.ts:146]: For my sanity, can we have a does not decode one of the other formats in this PR. Specifically 01 seems to match up the most

File: lib/plugins/Label_4A.ts [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+                time: DateTimeUtils.convertHHMMSSToTod(fields[2]),
+                timeFormat: 'tod',
+            };
+            decodeResult.raw.route = {waypoints: [wp1, wp2]};
+            decodeResult.formatted.items.push({
+                type: 'aircraft_route',
+                code: 'ROUTE',
+                label: 'Aircraft Route',
+                value: RouteUtils.routeToString(decodeResult.raw.route),
+            });
```
Comment (makrsmark) [inline lib/plugins/Label_4A.ts:64]: Can this move to ResultFormatter and update H1Helper?

File: lib/plugins/Label_5Z_Slash.test.ts [hunk 6]
```diff
@@ -160,7 +160,7 @@ describe('Label 5Z', () => {
   test('/C3 Request', () => {
     const text = '/C3 GATE REQ       / KBNA KEWR 22 115400 0554 ---- ---- ---- ----';
     const decodeResult = plugin.decode({ text: text });
-    console.log(JSON.stringify(decodeResult, null, 2));
```
Comment (makrsmark) [inline lib/plugins/Label_5Z_Slash.test.ts:163]: My bad, thank you!

File: lib/utils/coordinate_utils.ts [hunk 7]
```diff
@@ -73,4 +73,4 @@ export class CoordinateUtils {
   public static dmsToDecimalDegrees(degrees: number, minutes: number, seconds: number ) : number {
```
Comment (makrsmark) [inline lib/utils/coordinate_utils.ts:73]: I think this exists elsewhere

File: lib/utils/route_utils.ts [hunk 8]
```diff
@@ -4,6 +4,18 @@ import { Waypoint } from "../types/waypoint";
 import { CoordinateUtils } from "./coordinate_utils";
 
 export class RouteUtils {
+
+    public static formatFlightState(state: string): string {
```
Comment (makrsmark) [inline lib/utils/route_utils.ts:8]: Awesome!

**Review by makrsmark — COMMENTED** [review]
The only blocker is I'm 99% sure converting DMS to decimal degrees already exists. We should consolidate on one. I'll find where when I get in front of an IDE

File: lib/utils/coordinate_utils.ts (hunk 7 above)
Comment (makrsmark) [reply lib/utils/coordinate_utils.ts:73]: nvm - i can't see color (or lack of)

**Review by makrsmark — APPROVED** [review]
Will merge later today

**Comment by rpatel3001** [comment]
gonna add the does not decode tests quickly then address the C band and H1 helper things in new PRs

**Comment by rpatel3001** [comment]
Added a DIS01 message as the invalid test since it's also comma separated.

**Pull request merged**